src/Uno.UI/UI/Xaml/Markup/Reader/XamlObjectBuilder.cs

```diff
@@ -90,7 +92,17 @@ public XamlObjectBuilder(XamlFileDefinition xamlFileDefinition)
 
 			if (createInstanceFromXClass && TypeResolver.FindType(classMember?.Value?.ToString()) is { } classType)
 			{
-				return Activator.CreateInstance(classType);
+				var created = Activator.CreateInstance(classType);
+
+				if (created is FrameworkElement fe)
+				{
+					fe.SetBaseUri(fe.BaseUri.ToString(), _fileUri, control.LineNumber, control.LinePosition);
```

Comment:
Do we need to do this even when `_fileUri` is empty? I don't really know.

Reply:
I've added checks for not null or whitespace.
There are other checks for if it's blank when setting, but adding the extra check here is clearer.